src/compile/header/index.ts

```diff
@@ -75,7 +75,7 @@ export function getTitleGroup(model: Model, channel: HeaderChannel) {
     // also make sure that guide-title config override these Vega-lite default
   };
 
-  getHeaderTitleProperties(config, update);
+  getHeaderConfigProperties(config, update);
```

Comment:
This method is causing side effect.  Perhaps it's better to not pass update in and use